ktor-utils/jvm/test/io/ktor/tests/utils/DropLeadingTopDirsTest.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2014-2019 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.tests.utils
+
+import io.ktor.util.*
+import kotlin.test.*
+
+class DropLeadingTopDirsTest {
+    @Test
+    fun testEmptyPath() {
+        assertEquals(0, dropLeadingTopDirs(""))
+        assertEquals(0, dropLeadingTopDirs(" "))
+        assertEquals(1, dropLeadingTopDirs("/"))
+        assertEquals(1, dropLeadingTopDirs("\\"))
+    }
+
+    @Test
+    fun testSingleDot() {
+        assertEquals(0, dropLeadingTopDirs("."))
+        assertEquals(2, dropLeadingTopDirs("./"))
+        assertEquals(2, dropLeadingTopDirs(".\\"))
+    }
+
+    @Test
+    fun testSingleTopDir() {
+        assertEquals(2, dropLeadingTopDirs(".."))
+        assertEquals(3, dropLeadingTopDirs("../a"))
+        assertEquals(3, dropLeadingTopDirs("..\\a"))
+    }
+
+    @Test
+    fun testDoubleTopDir() {
+        assertEquals(5, dropLeadingTopDirs("../.."))
+        assertEquals(5, dropLeadingTopDirs("..\\.."))
+        assertEquals(6, dropLeadingTopDirs("../../"))
```

Comment:
Do we handle `/` and `\` similarly for all platforms? For UNIX it's OK to have a file with `\` in name:

```
$ cat > ..\..
hello
^C
$ cat ..\..
hello
```

Reply:
We probably could but for simplicity and with lack of real use-cases, it is a common pattern to treat both as path separators.